### PR TITLE
Migrate ordinary events to the new AMQP API

### PR DESCRIPTION
**Pull request by netoax**
**Describe what this PR introduces:**

This patch set continues migrating the babeltower to thew new AMQP API and it includes all operations besides `auth` and `list`.

related to #59 

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bug fix
- [ ] Feature
- [ ] Code style update
- [x] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce any breaking changes?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] Related issues are referenced in the PR (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing
- [ ] New/updated tests are included

**Testing environment:**

- Operating System/Platform: macOS Darwin.
- Go version: 1.14

### CODE REVIEW

File: pkg/server/handler.go [hunk 1]
```diff
@@ -73,20 +68,20 @@ func (mc *MsgHandler) subscribeToMessages(msgChan chan network.InMsg) error {
...
-
-	// Subscribe to messages received from the connector service
-	subscribe(msgChan, queueNameConnOut, exchangeConnOut, "topic", bindingKeyDataCommands)
-	subscribe(msgChan, queueNameConnOut, exchangeConnOut, "topic", bindingKeyDevice)
+	// Subscribe to general direct commands
+	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeyRegisterDevice)
+	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeyUnregisterDevice)
+	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeyRequestData)
+	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeyUpdateData)
+	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeySchemaSent)
```
Comment (lcbm) [inline pkg/server/handler.go]: Same as [here](https://github.com/CESARBR/knot-babeltower/pull/66#discussion_r417625667), regarding `"direct"`. Maybe this should be a `const` :smile:

File: pkg/server/handler.go [hunk 2]
```diff
@@ -73,20 +68,20 @@ func (mc *MsgHandler) subscribeToMessages(msgChan chan network.InMsg) error {
...
 
 	// Subscribe to request-reply messages received from any client
 	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeyAuthDevice)
 	subscribe(msgChan, queueNameCommands, exchangeDevices, "direct", bindingKeyListDevices)
 
+	// Subscribe to broadcasted data events
+	subscribe(msgChan, queueNameEvents, exchangeDataSent, "fanout", "")
+
```
Comment (lcbm) [inline pkg/server/handler.go]: Same as the previous (also [here](https://github.com/CESARBR/knot-babeltower/pull/66#discussion_r417625667)), regarding `"fanout"`. Maybe this should be a `const` :smile:

File: pkg/thing/delivery/amqp/client.go [hunk 3]
```diff
@@ -142,6 +143,17 @@ func (cs *commandSender) SendListResponse(things []*entities.Thing, corrID strin
...
 
+// SendUpdateData send update data command
+func (mp *msgClientPublisher) SendPublishedData(thingID string, data []entities.Data) error {
+	resp := &network.DataUpdate{ID: thingID, Data: data}
+	msg, err := json.Marshal(resp)
+	if err != nil {
+		return fmt.Errorf("message parsing error: %w", err)
+	}
+
+	return mp.amqp.PublishPersistentMessage("data.published", "fanout", "", msg)
```
Comment (lcbm) [inline pkg/thing/delivery/amqp/client.go]: Same as mentioned earlier, as an attempt to avoid hard-coded exchanges, topics, queues, etc.

**Review by lcbm — CHANGES_REQUESTED** [review]
Awesome patch! :rocket: 

I made a few remarks, and I also notice some typos in the commit messages :point_down: 

- commit 09592e63b9d6187821560523c46d9fb91c314ddb: `compatilize` is actually `make compatible` (I made the same mistake, hehe. Apparently, `compatibility` is not a word!
- commit 327a6c525a96983034891d6d407a821cc98a6c67: same as above, regarding`compatibilize` and also `suport` is supposed to be `supported`.
- commit 2524866148e46a1ad188feef54e49415695dbbe1: `updateed` should be `updated`

File: pkg/server/handler.go (hunk 1 above)
Comment (netoax) [reply pkg/server/handler.go]: Done.

File: pkg/server/handler.go (hunk 2 above)
Comment (netoax) [reply pkg/server/handler.go]: Done.

File: pkg/thing/delivery/amqp/client.go (hunk 3 above)
Comment (netoax) [reply pkg/thing/delivery/amqp/client.go]: Done.

**Review by lcbm — APPROVED** [review]
This is awesome, congrats! 🎊 🚀 💯 

I'm approving this in advance, but please don't forget to fix this 👇 😄 

These two identifiers are never used, but instead of deleting them we should add them to where their values are used...
- `updateDataOutKey` ("data.update"): add `updateDataOutKey` to **line 122**; 
- `requestDataOutKey` ("data.request"): add `requestDataOutKey` to **line 110**.

**Comment by netoax** [comment]
> This is awesome, congrats! 🎊 🚀 💯
> 
> I'm approving this in advance, but please don't forget to fix this 👇 😄
> 
> These two identifiers are never used, but instead of deleting them we should add them to where their values are used...
> 
> * `updateDataOutKey` ("data.update"): add `updateDataOutKey` to **line 122**;
> * `requestDataOutKey` ("data.request"): add `requestDataOutKey` to **line 110**.

Are you suggesting rename the local `routingKey` variables to `updateDataOutKey ` and `requestDataOutKey ` or concatenate them to the new strings?

**Comment by netoax** [comment]
@lcbm fixed.
- [requestDataOutKey](https://github.com/CESARBR/knot-babeltower/pull/67/commits/df984188eeee9f3c4e05f05d94ac6c7f7e3fabb8#diff-e0f5cd6a36757114487201738a4976e0R110)
- [updateDataOutKey](https://github.com/CESARBR/knot-babeltower/pull/67/commits/df984188eeee9f3c4e05f05d94ac6c7f7e3fabb8#diff-e0f5cd6a36757114487201738a4976e0R122)
- 2b9885cacff7ad80a04f480850d271a3284c6497

**Review by lcbm — APPROVED** [review]
LGTM! :smile:

File: pkg/server/handler.go [hunk 4]
```diff
@@ -76,20 +70,20 @@ func (mc *MsgHandler) subscribeToMessages(msgChan chan network.InMsg) error {
...
+	subscribe(msgChan, queueNameCommands, exchangeDevices, exchangeDevicesType, bindingKeyRequestData)
+	subscribe(msgChan, queueNameCommands, exchangeDevices, exchangeDevicesType, bindingKeyUpdateData)
+	subscribe(msgChan, queueNameCommands, exchangeDevices, exchangeDevicesType, bindingKeySchemaSent)
 
 	// Subscribe to request-reply messages received from any client
 	subscribe(msgChan, queueNameCommands, exchangeDevices, exchangeDevicesType, bindingKeyAuthDevice)
 	subscribe(msgChan, queueNameCommands, exchangeDevices, exchangeDevicesType, bindingKeyListDevices)
 
+	// Subscribe to broadcasted data events
+	subscribe(msgChan, queueNameEvents, exchangeDataSent, exchangeDataSentType, "")
```
Comment (ramonhpr) [inline pkg/server/handler.go]:
```suggestion
	subscribe(msgChan, queueNameEvents, exchangeDataSent, exchangeDataSentType, bindKeyAny)
```
Add other const value

**Review by esdrasjnr — APPROVED** [review]
Everything LGTM! Awesome patch!

File: pkg/server/handler.go [hunk 5]
```diff
@@ -13,25 +13,19 @@ import (
...
+	exchangeDataSentType       = "fanout"
+	queueNameCommands          = "babeltower-command-messages"
+	queueNameEvents            = "babeltower-event-messages"
+	bindingKeyAuthDevice       = "device.auth"
+	bindingKeyListDevices      = "device.list"
+	bindingKeyRegisterDevice   = "device.register"
+	bindingKeyUnregisterDevice = "device.unregister"
+	bindingKeyRequestData      = "data.request"
+	bindingKeyUpdateData       = "data.update"
+	bindingKeySchemaSent       = "device.schema.sent"
```
Comment (ramonhpr) [inline pkg/server/handler.go]:
```suggestion
	exchangeDevices            = "device"
	exchangeDevicesType        = "direct"
	exchangeDataSentType       = "fanout"
	exchangeDataSent           = "data.sent"
	queueNameCommands          = "babeltower-command-messages"
	queueNameEvents            = "babeltower-event-messages"
	bindingKeyAuthDevice       = "device.auth"
	bindingKeyListDevices      = "device.list"
	bindingKeyRegisterDevice   = "device.register"
	bindingKeyUnregisterDevice = "device.unregister"
	bindingKeyRequestData      = "data.request"
	bindingKeyUpdateData       = "data.update"
	bindingKeySchemaSent       = "device.schema.sent"
```

**Review by ramonhpr — COMMENTED** [review]
LGTM, just see my suggestion about binding key

File: pkg/server/handler.go (hunk 4 above)
Comment (netoax) [reply pkg/server/handler.go]: Done. I found `bindingKeyEmpty` better.

File: pkg/server/handler.go (hunk 5 above)
Comment (netoax) [reply pkg/server/handler.go]: Done.

**Pull request merged**